src/parser.lalrpop

```diff
@@ -216,6 +217,10 @@ ElseBlock: Vec<Statement> = {
 	"else" "{" <Stmt+> "}" => <>
 }
 
+OptionalBlock: OptionalBlock = {
+	"optional" "{" <Stmt*> "}" => OptionalBlock::new(<>)
```

Comment:
I am guessing * is 0 to many.  Do we want to allow for an empty optional block?

Reply:
Two initial thoughts:

1. I think it might be desirable if someone wants to comment out all the rules in an optional block?
2. If we do want to ban them, the parser level may not be the best place.  It might make more sense to parse, and then report an error when actually processing the block in ValidatedStatement::new()

Reply:
1. It would only be a few more lines to comment out the block itself.  So, I don't weigh that too heavily.
2. My initial thought would be less stuff to deal with past the parser, but I do agree the parser error is going to be less meaningful than one we put there ourselves.

One last thought is just that an empty optional block is meaningless.  Other empty blocks we have may not be.

I don't feel too passionately one way or another, so I will leave the call up to you.

Reply:
Having sat on this for a little while, what I would like to do is allow empty optional blocks.  I don't see it hurting anything, and I think the commenting out scenario is not unreasonable.  Perhaps it's something we might warn or lint on in the future.

As an aside, I'd generally prefer a more permissive parser with errors later as a non-strict general principle.  Since if we report errors after parsing, we can be more specific and helpful in the error message.